♻️ Refactor `load_understat_league_shots()`

nepito opened the pull request:
- Add variable `LEAGUES` to  change **if** statement by **list**.
- Extract function `check_league_name()`.

tonyelhabr commented:
> * Add variable `LEAGUES` to  change **if** statement by **list**.
> 
>     * Extract function `check_league_name()`.

Could you call `check_league_name()` in `understat_league_season_shots` too?

JaseZiv commented:
> * Add variable `LEAGUES` to  change **if** statement by **list**.
> * Extract function `check_league_name()`.

Thanks for the contribution.

I was wondering what the purpose of this change is though? If it's to minimise the chance users enter the wrong league name, I don't know how this will solve that problem. A solution might be as simple as an update to the roxygen notes in each of the functions?

nepito commented:
> I was wondering what the purpose of this change is though? If it's to minimise the chance users enter the wrong league name, I don't know how this will solve that problem. A solution might be as simple as an update to the roxygen notes in each of the functions?

Hi @JaseZiv  is my placer. 

The **if** conditionals are contraindicated. I think the **lists** are a bit clearer. It's just a matter of taste. Errors can continue to be made by users.

Although it is not easy to see, the ultimate goal is to reduce the time when running the tests. If we can test that piece of code, it will not be necessary to communicate over the internet; perhaps the tests will run faster.

nepito commented:
> > * Add variable `LEAGUES` to  change **if** statement by **list**.
> >   
> >   * Extract function `check_league_name()`.
> 
> Could you call `check_league_name()` in `understat_league_season_shots` too?

Hi @tonyelhabr , I tried call `check_league_name()` in `understat_league_season_shots()`, but the test took a long time. I think are too much data. So, I applied your suggestion in `understat_league_match_results()`.

tonyelhabr commented on R/load_understat.R:
```diff
@@ -50,3 +51,8 @@ load_understat_league_shots <- function(league) {
 }
+
+check_league_name <- function(league_name) {
+  leagues <- names(LEAGUES)
+  if(!league_name %in% leagues) stop("Check league name")
```
```suggestion
  arg_match0(league_name, names(LEAGUES))
```

tonyelhabr replied:
If we're going to improve the error handling, then we should really improve it. `arg_match0` specifies what the allowed values are when erroring.

nepito replied:
The `arg_match0()` function is great. Thank you very much for recommending it 🙌🏿.

tonyelhabr commented on tests/testthat/test-understat.R:
```diff
@@ -47,3 +48,6 @@ test_that("understat_team_stats_breakdown() works", {
   expect_gt(nrow(team_stats), 0)
 })
 
+test_that("understat_league_season_shots() works", {
```
you can just delete this. `check_league_name` is already directly tested and indirectly tested

nepito replied:
Done :+1: .

JaseZiv commented:
Hi @nepito,

As part of this PR, can you also increment the version number in `DESCRIPTION` to `0.6.2.2000` and update the `NEWS.md` file also.

Please follow past commits to those files for examples.

Let me know if you have any questions.

Thanks

nepito commented:
> incremen

Hi @JaseZiv,
The current version is 0.6.2.2000; must I increase it to 0.6.2.3000?

JaseZiv commented:
> > incremen
> 
> 
> 
> Hi @JaseZiv,
> 
> The current version is [0.6.2.2000](https://github.com/JaseZiv/worldfootballR/blob/main/DESCRIPTION#L4); must I increase it to 0.6.2.3000?

Oh yes of course - sorry for the typo! Thanks

nepito commented:
> > > > incremen
> > > 
> > > 
> > > Hi @JaseZiv,
> > > The current version is [0.6.2.2000](https://github.com/JaseZiv/worldfootballR/blob/main/DESCRIPTION#L4); must I increase it to 0.6.2.3000?
> > 
> > 
> > Oh yes of course - sorry for the typo! Thanks
> 
> Hi @JaseZiv I added the information in the **Improvements** section but not in the development version. Is this correct?

Hi @JaseZiv,
I moved the information to the development version. Also, I formatted the files with Styler. What do you think?

nepito commented:
Hi @JaseZiv and @tonyelhabr 
I've fixed a bug in the tests. GitHub Actions should pass now. Any feedback will be appreciated :smiley:
